### PR TITLE
sent userlist message to guests

### PR DESCRIPTION
**Pull request by ejohnson44**

### CODE REVIEW

File: SoundStream/src/com/lastcrusade/soundstream/service/MessagingService.java [hunk 1]
```diff
@@ -50,6 +52,9 @@
     public static final String ACTION_LIBRARY_MESSAGE = MessagingService.class.getName() + ".action.LibraryMessage";
     public static final String EXTRA_SONG_METADATA    = MessagingService.class.getName() + ".extra.SongMetadata";
 
+    public static final String ACTION_NEW_CONNECTED_USERS = MessagingService.class.getName() + ".action.UserListMessage";
```
Comment (reidbaker) [inline SoundStream/src/com/lastcrusade/soundstream/service/MessagingService.java]: To match our convention lets have this be `ACTION_NEW_CONNECTED_USERS_MESSAGE`

File: SoundStream/src/com/lastcrusade/soundstream/CustomApp.java [hunk 2]
```diff
@@ -69,7 +71,17 @@ public void onReceiveAction(Context context, Intent intent) {
...
                     userList.removeUser(macAddress);
-                    userList.notifyUpdate(CustomApp.this);
+                    notifyUserListUpdate();
+                }
+            })
+            .addAction(MessagingService.ACTION_NEW_CONNECTED_USERS, new IBroadcastActionHandler() {
+                
+                @Override
+                public void onReceiveAction(Context context, Intent intent) {
+                    Log.i(TAG, "New Connected Users Message Recieved");
```
Comment (reidbaker) [inline SoundStream/src/com/lastcrusade/soundstream/CustomApp.java]: Probably dont want that to log forever. Lets remove it.

File: SoundStream/src/com/lastcrusade/soundstream/model/User.java [hunk 3]
```diff
@@ -1,8 +1,16 @@
...
+import com.lastcrusade.soundstream.util.DefaultParcelableCreator;
+
+import android.os.Parcel;
+import android.os.Parcelable;
 import android.util.Log;
 
-public class User {
+public class User implements Parcelable{
+    
+    public static final Parcelable.Creator<SongMetadata> CREATOR = new DefaultParcelableCreator(User.class);
```
Comment (reidbaker) [inline SoundStream/src/com/lastcrusade/soundstream/model/User.java]: Lets include @theJenix comment about why this is here an why we cant change it.

File: SoundStream/src/com/lastcrusade/soundstream/model/User.java [hunk 4]
```diff
@@ -17,6 +25,12 @@ public User(String bluetoothID, String macAddress, int color){
         this.color = color;
     }
     
+    public User(Parcel in){
```
Comment (reidbaker) [inline SoundStream/src/com/lastcrusade/soundstream/model/User.java]: No longer used  please remove

File: SoundStream/src/com/lastcrusade/soundstream/model/User.java [hunk 5]
```diff
@@ -32,4 +46,18 @@ public String getMacAddress(){
         return macAddress;
 
     }
+
+    @Override
+    public int describeContents() {
```
Comment (reidbaker) [inline SoundStream/src/com/lastcrusade/soundstream/model/User.java]: What does this do? Needs a comment :)

File: SoundStream/src/com/lastcrusade/soundstream/model/User.java [hunk 6]
```diff
@@ -32,4 +46,18 @@ public String getMacAddress(){
...
+        return 0;
+    }
+
+    @Override
+    public void writeToParcel(Parcel dest, int flags) {
```
Comment (reidbaker) [inline SoundStream/src/com/lastcrusade/soundstream/model/User.java]: Is this still needed?

File: SoundStream/src/com/lastcrusade/soundstream/model/UserList.java [hunk 7]
```diff
@@ -4,12 +4,18 @@
...
+import android.os.Parcelable;
 import android.util.Log;
 
 import com.lastcrusade.soundstream.util.BroadcastIntent;
+import com.lastcrusade.soundstream.util.DefaultParcelableCreator;
 
-public class UserList {
+public class UserList implements Parcelable{
 
+    public static final Parcelable.Creator<SongMetadata> CREATOR = new DefaultParcelableCreator(UserList.class);
```
Comment (reidbaker) [inline SoundStream/src/com/lastcrusade/soundstream/model/UserList.java]: comment here

File: SoundStream/src/com/lastcrusade/soundstream/model/UserList.java [hunk 8]
```diff
@@ -25,6 +31,14 @@ public UserList(String bluetoothID, String macAddress){
         this();
         connectedUsers.add(new User(bluetoothID, macAddress, userColors.getNextAvailableColor()));
     }
+    
+    public UserList(Parcel in){
+        this();
```
Comment (reidbaker) [inline SoundStream/src/com/lastcrusade/soundstream/model/UserList.java:37]: comment here that `this();` calls the main constructor.

File: SoundStream/src/com/lastcrusade/soundstream/model/UserList.java [hunk 9]
```diff
@@ -113,4 +123,20 @@ public String toString() {
         }
         return users;
     }
+
+    @Override
```
Comment (reidbaker) [inline SoundStream/src/com/lastcrusade/soundstream/model/UserList.java:129]: same as before

File: SoundStream/src/com/lastcrusade/soundstream/service/MessagingService.java [hunk 10]
```diff
@@ -295,4 +315,11 @@ public void sendStringMessage(String message) {
             Log.wtf(TAG, e);
         }
     }
+    
+    //sends the user list out to everyone
+    public void sendUserListMessage(UserList userlist){
+        UserListMessage ulm = new UserListMessage(userlist);
+        broadcastMessageToFans(ulm);
+        Log.i("Messaging Service", "User List Message Sent");
```
Comment (reidbaker) [inline SoundStream/src/com/lastcrusade/soundstream/service/MessagingService.java]: Lets use a TAG variable

File: SoundStream/src/com/lastcrusade/soundstream/service/MessagingService.java [hunk 11]
```diff
@@ -50,6 +52,9 @@
     public static final String ACTION_LIBRARY_MESSAGE = MessagingService.class.getName() + ".action.LibraryMessage";
     public static final String EXTRA_SONG_METADATA    = MessagingService.class.getName() + ".extra.SongMetadata";
 
+    public static final String ACTION_NEW_CONNECTED_USERS_MESSAGE = MessagingService.class.getName() + ".action.UserListMessage";
+    public static final String EXTRA_USER_LIST    = MessagingService.class.getName() + ".extra.UserList";
```
Comment (theJenix) [inline SoundStream/src/com/lastcrusade/soundstream/service/MessagingService.java]: In a future commit, please line up the = signs between these 2 lines...it makes this code easier to read.

File: SoundStream/src/com/lastcrusade/soundstream/net/message/UserListMessage.java [hunk 12]
```diff
@@ -10,11 +10,15 @@
 public class UserListMessage extends ADataMessage {
 	private final String TAG = UserListMessage.class.getName();
 	
-	private UserList userList = new UserList(); 
+	private UserList userList; 
 	
-	public UserListMessage() {}
+	public UserListMessage() {
```
Comment (theJenix) [inline SoundStream/src/com/lastcrusade/soundstream/net/message/UserListMessage.java]: Please make this constructor package protected, as it should only be called by the Messenger.  See ConnectFansMessage for a comment you can pull in to explain this.

**Pull request merged**